src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -32,11 +32,37 @@ public class LockFileBuilder
             _includeFlagGraphs = includeFlagGraphs;
         }
 
+        /// <summary>
+        /// Creates a LockFile from an existing one given a project, a retore graph and local repositories
+        /// </summary>
+        /// <param name="previousLockFile">Base Lock file</param>
+        /// <param name="project">A buildable, restorable project</param>
+        /// <param name="targetGraphs">Restore target graph</param>
+        /// <param name="localRepositories">V3 NuGet Sources collection</param>
+        /// <param name="context">Not used</param>
+        /// <returns>A LockFile created from arguments</returns>
+        [Obsolete("Use CreateLockFile() overload without context argument instead")]
```

Comment:
Same here, I want to avoid adding new API for theoretical purposes in very specific codepaths. 
We're at a major verison, I'd rather be removing APIs.